src/hooks/transactions/index.tsx

```diff
@@ -66,7 +66,8 @@ export function useIsPendingApproval(token?: Token): boolean {
 }
 
 export type OnTxSubmit = (hash: string, tx: Transaction) => void
-export type OnTxSuccess = (hash: string, receipt: TransactionReceipt) => void
+type WithRequired<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>
```

Comment:
this seems like it should go in a `types.d.ts` file so it can be used globally without needing to import 👀